src/core/lib/slice/slice.c

```diff
@@ -341,6 +348,40 @@ grpc_slice grpc_slice_split_tail(grpc_slice *source, size_t split) {
   return tail;
 }
 
+grpc_slice grpc_slice_split_tail_no_ref(grpc_slice *source, size_t split) {
+  grpc_slice tail;
+
+  if (source->refcount == NULL) {
+    /* inlined data, copy it out */
+    GPR_ASSERT(source->data.inlined.length >= split);
+    tail.refcount = NULL;
+    tail.data.inlined.length = (uint8_t)(source->data.inlined.length - split);
+    memcpy(tail.data.inlined.bytes, source->data.inlined.bytes + split,
+           tail.data.inlined.length);
+    source->data.inlined.length = (uint8_t)split;
+  } else {
+    size_t tail_length = source->data.refcounted.length - split;
+    GPR_ASSERT(source->data.refcounted.length >= split);
+    if (tail_length < sizeof(tail.data.inlined.bytes)) {
+      /* Copy out the bytes - it'll be cheaper than refcounting */
+      tail.refcount = NULL;
+      tail.data.inlined.length = (uint8_t)tail_length;
+      memcpy(tail.data.inlined.bytes, source->data.refcounted.bytes + split,
+             tail_length);
+    } else {
```

Comment:
This else block is the only segment in which the logic differs from grpc_slice_split_tail. Would rather see some structure that duplicates less code. Some ideas;
* an internal version that takes the refcount to assign to tail.refcount
* grpc_slice_split_tail takes in a bool, defaulted to true, which dictates if it will alter the refcounts.